pkg/block/metadata/compactormeta.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import "github.com/oklog/ulid"
+
+// CompactorMetaFilename is the known json filename to store compactor metadata.
+const CompactorMetaFilename = "compactor-meta.json"
```

Comment:
I am not sure about this `compactor` meta file naming. It is tied to the source of this file, who created it. What if someday we would remove compactor and allow something else to delete blocks? Like a repair job? 

What about naming this in a way that is more related to purpose. Like `deletion-meta.json` or `post-meta.json` ... not sure. @pstibrany @pracucci @squat @brancz  any thoughts?

Reply:
I think I like `deletion-meta.json` best.

Reply:
Or `deletion-mark.json`

Reply:
> Or `deletion-mark.json`

👍 – makes its purpose clear